packages/backend/server/src/plugins/copilot/providers/gemini.ts

```diff
@@ -43,22 +47,36 @@ export class GeminiProvider
   override readonly models = [
     // text to text
     'gemini-2.0-flash-001',
-    'gemini-2.5-pro-preview-03-25',
+    'gemini-2.5-flash-preview-04-17',
+    'gemini-2.5-pro-preview-05-06',
     // embeddings
     'text-embedding-004',
   ];
 
-  #instance!: GoogleGenerativeAIProvider;
+  private readonly MAX_STEPS = 20;
+
+  private readonly CALLOUT_PREFIX = '\n> [!]\n> ';
+
+  #instance!: GoogleVertexProvider;
 
   override configured(): boolean {
-    return !!this.config.apiKey;
+    return !!this.config.apiKey && !!this.config.privateKey;
   }
 
   protected override setup() {
     super.setup();
-    this.#instance = createGoogleGenerativeAI({
-      apiKey: this.config.apiKey,
-      baseURL: this.config.baseUrl,
+
+    this.#instance = createVertex({
+      project: this.config.projectId ?? 'affine-us',
+      location: this.config.location ?? 'us-central1',
```

Comment:
these two values should not have default values
need to be validated for existence in `configured()`

Reply:
Thanks for your review suggestion! Code Updated.